src/hotspot/cpu/x86/jeandleRuntimeRoutine_x86.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2025, the Jeandle-JDK Authors. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "jeandle/jeandleRuntimeRoutine.hpp"
+
+#include "utilities/debug.hpp"
+#include "asm/macroAssembler.hpp"
+#include "asm/macroAssembler.inline.hpp"
+#include "code/codeBlob.hpp"
+#include "memory/resourceArea.hpp"
+#include "oops/oopsHierarchy.hpp"
+#include "runtime/frame.hpp"
+#include "runtime/interfaceSupport.inline.hpp"
+
+#define __ masm->
+
+// When a Jeandle compiled method throwing an exception, patch its return address to exceptional_exit blob.
+JRT_ENTRY(void, JeandleRuntimeRoutine::install_exceptional_return(oopDesc* exception, JavaThread* current))
+  assert(oopDesc::is_oop(exception), "must be a valid oop");
+  RegisterMap r_map(current,
+                    RegisterMap::UpdateMap::skip,
+                    RegisterMap::ProcessFrames::include,
+                    RegisterMap::WalkContinuation::skip);
+  frame exception_frame = current->last_frame().sender(&r_map);
+  CodeBlob* exception_code = exception_frame.cb();
+  guarantee(exception_code != nullptr && exception_code->is_compiled_by_jeandle(), "install_exceptional_return must be jumped from jeandle compiled method");
+
+  intptr_t* sender_sp = exception_frame.unextended_sp() + exception_code->frame_size();
+
+  address* return_address = (address*)(sender_sp - 1);
+
+  current->set_exception_pc(*return_address);
+  current->set_exception_oop(exception);
+
+  // Change the return address to exceptional return blob.
+  *return_address = _routine_entry[_exceptional_return];
+JRT_END
+
+// When a Jeandle compiled method throwing an exception, its return address
+// will be patched to this blob. Here we find the right exception handler,
+// then jump to.
+// The exception oop and the exception pc have been set by
+// JeandleRuntimeRoutine::install_exceptional_return.
```

Comment:
Maybe it's better to add some comments for better understanding. Like the register conventions for exception handling bolo:
```cpp
Arguments:
  rax: exception oop
  rdx: exception pc

Results:
  rax: ...
  rdx: ...
```